detox/src/runners/jest/DetoxJestAdapter.js

```diff
@@ -0,0 +1,63 @@
+class DetoxJestAdapter /* implements JasmineReporter */ {
+  constructor(detox) {
+    this.detox = detox;
+    this._currentSpec = null;
+    this._todos = [];
+    this.timeout = 30000;
+  }
+
+  async beforeEach() {
+    await this._flush();
+    await this.detox.beforeEach(this._currentSpec);
+  }
+
+  async afterAll() {
+    await this._flush();
+  }
+
+  async _afterEach(previousSpec) {
+    await this.detox.afterEach(previousSpec);
+  }
+
+  async _flush() {
+    const t = this._todos;
+
+    while (t.length > 0) {
+      await Promise.resolve().then(t.shift()).catch(()=>{});
+    }
+  }
+
+  specStarted(result) {
+    if (result.pendingReason) {
+      return;
+    }
+
+    const spec = Object.freeze({
```

Comment:
This object is going inside Detox implementation, which just reads from it (correct me if I'm wrong please)
Why do we need to freeze it ?

Reply:
I suggest we get rid of all `Object.freeze`, especially if the consumers are us.